webapp/src/components/admin_settings/job_table/job_schedule_modal.tsx

```diff
@@ -21,14 +23,29 @@ const styles = {
     },
 };
 
-const JobScheduleModal = ({show, onClose, onSubmit}: Props) => {
+const JobScheduleModal = ({show, min_t, max_t, onClose, onSubmit}: Props) => {
     const today = new Date();
-    const defaultSelected: DateRange = {
+    let defaultSelected: DateRange = {
         from: addDays(today, -3),
         to: today,
     };
-    const [range, setRange] = useState<DateRange | undefined>(defaultSelected);
 
+    let fromDate = addDays(today, -14);
+    let toDate = today;
+    if (min_t && max_t) {
+        fromDate = new Date(min_t);
+        toDate = new Date(max_t);
+
+        const oneDay = 24 * 60 * 60 * 1000; // hours*minutes*seconds*milliseconds
+        if (Math.round(Math.abs((max_t - min_t) / oneDay)) <= 3) {
+            defaultSelected = {
+                from: fromDate,
+                to: toDate,
+            };
+        }
+    }
+
+    const [range, setRange] = useState<DateRange | undefined>(defaultSelected);
```

Comment:
Unless you're explicitly using `setRange(undefined)` later, you can use:
```suggestion
    const [range, setRange] = useState(defaultSelected);
```

Reply:
Ah, I see, thanks!